src/sass/components/_footer.scss

```diff
@@ -58,39 +63,75 @@
 
   @media (min-width: $tablet-screen) {
     // Name the footer areas.
-    .footer__branding { grid-area: branding; }
-    .footer__social { grid-area: social; }
-    .footer__about-text { grid-area: about; }
-    .footer__contact { grid-area: contact; }
-    .footer__menu--wrapper { grid-area: menu_wrapper; }
-    .footer__bottom-links { grid-area: links; }
+    .footer__branding {
+      grid-area: branding;
+    }
+    .footer__social {
+      grid-area: about;
+      align-self: end;
+      .social-icon {
+        padding-bottom: 0;
+      }
+    }
+    .footer__about-text {
+      grid-area: social;
+    }
+    .footer__contact {
+      grid-area: about;
+    }
+    .footer__menu--wrapper {
+      grid-area: social_wrapper;
+      .footer__menu--list {
+        margin-top: 0;
+        .child-menu--wrapper {
+          padding-top: .5rem;
+        }
+      }
+    }
+    .footer__bottom-links {
+      grid-area: about_wrapper;
+      p {
+        margin: 4px 0;
+      }
+    }
 
     .grid-container {
       grid-template-columns: 1fr 40%;
       grid-template-rows: auto;
-      grid-template-areas: 
-      "branding branding"
-      "social social"
-      "about menu_wrapper"
-      "contact menu_wrapper"
-      "links menu_wrapper";
+      grid-template-areas:
+      "branding branding_wrapper"
```

Comment:
Minor change requests: I don't see where branding_wrapper is defined. This can be set to `branding branding` if it spans 2 columns.
Also the 2nd items can be named in a descriptive manner to help with future troubleshooting. eg. the `social social_wrapper` grid-areas can be renamed to `footer_text menu`.

Both are optional and don't need to block merging this so I'll set the PR to approved and leave the actual merge up to you @jaube05